test/unit/math/prim/fun/fft_test.cpp

```diff
@@ -0,0 +1,208 @@
+}
+
+TEST(primFun, inv_fft) {
+  typedef std::complex<double> c_t;
+  typedef Eigen::Matrix<std::complex<double>, -1, 1> cv_t;
+  using stan::math::inv_fft;
+
+  // reference answers calculated using Scipy.fft with double precision
+
+  cv_t y0(0);
+  cv_t x0 = inv_fft(y0);
+  EXPECT_EQ(0, x0.size());
+
+  cv_t y1(1);
+  y1 << c_t(-3.247, 1.98555);
+  cv_t x1 = inv_fft(y1);
+  EXPECT_EQ(1, x1.size());
+  EXPECT_EQ(real(x1[0]), -3.247);
+  EXPECT_EQ(imag(x1[0]), 1.98555);
+
+  cv_t y(3);
+  y << c_t(-9, 14), c_t(0.80384758, -13.46410162),
+      c_t(11.19615242, -6.53589838);
+  cv_t x = inv_fft(y);
+  EXPECT_EQ(3, y.size());
+  EXPECT_NEAR(real(x[0]), 1, 1e-6);
+  EXPECT_NEAR(imag(x[0]), -2, 1e-6);
+  EXPECT_NEAR(real(x[1]), -3, 1e-6);
+  EXPECT_NEAR(imag(x[1]), 5, 1e-6);
+  EXPECT_NEAR(real(x[2]), -7, 1e-6);
+  EXPECT_NEAR(imag(x[2]), 11, 1e-6);
+}
+
+TEST(primFun, fft2) {
+  typedef std::complex<double> c_t;
+  typedef Eigen::Matrix<std::complex<double>, -1, -1> cm_t;
+  using stan::math::fft2;
+  using stan::math::inv_fft2;
+
+  // reference answers calculated using Scipy.fft with double
+  // precision, and verified with R
+
+  cm_t x(0, 0);
+  cm_t y = fft2(x);
+  EXPECT_EQ(0, y.rows());
+  EXPECT_EQ(0, y.cols());
+
+  cm_t x11(1, 1);
+  x11 << c_t(1.0, -3.9);
+  cm_t y11 = fft2(x11);
+  EXPECT_NEAR(1.0, std::real(y11(0, 0)), 1e-6);
+  EXPECT_NEAR(-3.9, std::imag(y11(0, 0)), 1e-6);
+
+  cm_t x12(1, 2);
+  x12 << c_t(1.0, -3.9), c_t(-8.6, 0.2);
+  cm_t y12 = fft2(x12);
+  EXPECT_EQ(1, y12.rows());
+  EXPECT_EQ(2, y12.cols());
+  EXPECT_NEAR(-7.6, real(y12(0, 0)), 1e-6);
+  EXPECT_NEAR(-3.7, imag(y12(0, 0)), 1e-6);
+  EXPECT_NEAR(9.6, real(y12(0, 1)), 1e-6);
+  EXPECT_NEAR(-4.1, imag(y12(0, 1)), 1e-6);
+
+  cm_t x33(3, 3);
+  x33 << c_t(1, 2), c_t(3, -1.4), c_t(2, 1), c_t(3, -9), c_t(2, -1.3),
+      c_t(3.9, -1.8), c_t(13, -1.8), c_t(1.3, 1.9), c_t(-2.2, -2.2);
+  cm_t y33 = fft2(x33);
+  EXPECT_EQ(3, y33.rows());
+  EXPECT_EQ(3, y33.cols());
+
+  // check vs. results from R (matches SciPy)
+  EXPECT_NEAR(27, real(y33(0, 0)), 1e-6);
+  EXPECT_NEAR(-12.6, imag(y33(0, 0)), 1e-6);
+  EXPECT_NEAR(13.90525589, real(y33(0, 1)), 1e-6);
+  EXPECT_NEAR(-9.15166605, imag(y33(0, 1)), 1e-6);
+  EXPECT_NEAR(10.09474411, real(y33(0, 2)), 1e-1);
+  EXPECT_NEAR(-4.64833395, imag(y33(0, 2)), 2e-1);
+  EXPECT_NEAR(-13.160254038, real(y33(1, 0)), 1e-1);
+  EXPECT_NEAR(11.471281292, imag(y33(1, 0)), 2e-1);
+  EXPECT_NEAR(-13.29326674, real(y33(1, 1)), 1e-1);
+  EXPECT_NEAR(20.88153533, imag(y33(1, 1)), 1e-1);
+  EXPECT_NEAR(-13.25262794, real(y33(1, 2)), 1e-1);
+  EXPECT_NEAR(15.82794549, imag(y33(1, 2)), 1e-1);
+  EXPECT_NEAR(4.160254038, real(y33(2, 0)), 1e-1);
+  EXPECT_NEAR(5.928718708, imag(y33(2, 0)), 1e-1);
+  EXPECT_NEAR(-11.34737206, real(y33(2, 1)), 1e-1);
+  EXPECT_NEAR(-7.72794549, imag(y33(2, 1)), 1e-1);
+  EXPECT_NEAR(4.89326674, real(y33(2, 2)), 1e-1);
+  EXPECT_NEAR(-1.98153533, imag(y33(2, 2)), 1e-1);
```

Comment:
[optional] I think this would be easier to read if the correct answers were put into a matrix and just looped over

Reply:
fixed as suggested.